internal/config/skip.go

```diff
@@ -4,47 +4,90 @@ import (
 	"github.com/gobwas/glob"
 
 	"github.com/evilmartians/lefthook/internal/git"
+	"github.com/evilmartians/lefthook/internal/log"
 )
 
-func doSkip(gitState git.State, skip, only interface{}) bool {
+type SkipChecker struct {
+	Executor Exec
+}
+
+func NewSkipChecker(executor Exec) *SkipChecker {
+	if executor == nil {
+		executor = NewOsExec()
+	}
+
+	return &SkipChecker{Executor: executor}
+}
+
+func (sc *SkipChecker) DoSkip(gitState git.State, skip interface{}, only interface{}) bool {
```

Comment:
Let's change the name of this method? I think for the `SkipChecker` it's more intuitive to use `Check` method. WDYT?